inc/com/centreon/engine/host.hh

```diff
@@ -153,6 +160,29 @@ class                 host : public notifier {
   int                is_escalated_contact(contact* cntct);
   bool               is_result_fresh(time_t current_time, int log_this);
 
+  int                run_sync_host_check_3x(enum host::host_state* check_result_code,
+                                            int check_options,
+                                            int use_cached_result,
+                                            unsigned long check_timestamp_horizon);
+  int                perform_on_demand_host_check_3x( enum host::host_state* check_result_code,
+                                            int check_options,
+                                            int use_cached_result,
+                                            unsigned long check_timestamp_horizon);
+
+  int                perform_on_demand_host_check(enum host::host_state* check_return_code,
+                                                 int check_options,
+                                                 int use_cached_result,
+                                                 unsigned long check_timestamp_horizon);
+  int                process_host_check_result_3x(enum host::host_state new_state,
+                                                 char* old_plugin_output,
+                                                 int check_options,
+                                                 int reschedule_check,
+                                                 int use_cached_result,
+                                                 unsigned long check_timestamp_horizon);
+  enum host_state    determine_host_reachability();
```

Comment:
idem, if we are in the host class, this method could be named determine_reachability().